### ISSUE TITLE
[Issue]: suddenly very bad performance

### ISSUE THREAD

**Issue by CapsAdmin** [vladmandic/automatic#155]
### Issue Description

I'm not sure if I can blame updates on this one, at first I thought it was my pc so I checked out A1111 and it works fine there. It is using the GPU but something is off.

Usually a normal gen takes 1.5 seconds, but now it takes around 20 seconds

I tried running with --profile but I'm not sure if I see anything specific. 

```
Profile:          335428 function calls (304303 primitive calls) in 19.364 seconds

   Ordered by: cumulative time
   List reduced from 907 to 15 due to restriction <15>

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    0.000    0.000   19.364   19.364 /home/caps/projects/automatic/modules/call_queue.py:24(f)
        1    0.000    0.000   19.363   19.363 /home/caps/projects/automatic/modules/txt2img.py:10(txt2img)
        1    0.000    0.000   19.363   19.363 /home/caps/projects/automatic/modules/processing.py:489(process_images)
        1    0.001    0.001   19.363   19.363 /home/caps/projects/automatic/modules/processing.py:526(process_images_inner)
  8252/16    0.019    0.000   19.045    1.190 /home/caps/.local/lib/python3.10/site-packages/torch/nn/modules/module.py:1494(_call_impl)
        3    0.000    0.000   19.005    6.335 /home/caps/.local/lib/python3.10/site-packages/torch/utils/_contextlib.py:112(decorate_context)
   632/12    0.001    0.000   18.988    1.582 /home/caps/projects/automatic/modules/sd_hijack_utils.py:17(<lambda>)
   632/12    0.001    0.000   18.988    1.582 /home/caps/projects/automatic/modules/sd_hijack_utils.py:24(__call__)
     1060    0.001    0.000   18.450    0.017 /home/caps/projects/automatic/extensions-builtin/a1111-sd-webui-lycoris/lycoris.py:740(lyco_Conv2d_forward)
     1060    0.001    0.000   18.355    0.017 /home/caps/projects/automatic/extensions-builtin/Lora/lora.py:316(lora_Conv2d_forward)
     1060    0.002    0.000   18.281    0.017 /home/caps/.local/lib/python3.10/site-packages/torch/nn/modules/conv.py:462(forward)
     1060    0.002    0.000   18.278    0.017 /home/caps/.local/lib/python3.10/site-packages/torch/nn/modules/conv.py:454(_conv_forward)
     1060   18.276    0.017   18.276    0.017 {built-in method torch.conv2d}
        1    0.000    0.000   17.622   17.622 /home/caps/projects/automatic/modules/processing.py:868(sample)
        1    0.000    0.000   17.621   17.621 /home/caps/projects/automatic/modules/sd_samplers_kdiffusion.py:330(sample)
```

Not sure what to make of it. 
I don't see anything related to sd_hijack_optimizations.py anywhere? (though it says the results are reduced). `{built-in method torch.conv2d}` also seems worrying, as if it's outside of python's control.

I tried resetting everything except removing venv, I tried reinstalling torch, I tried disabling lora and lycoris. Maybe if I nuke everything and reinstall it might work but I'm thinking maybe it's better to report and investigate.

I also tried different cross attention optimizations (and making sure to switch model back and forth) but they're all the same.

### Version Platform Description

arch: x86_64
cpu: x86_64
system: Linux
release: 6.0.10
python: 3.10.6
device: AMD Radeon RX 6900 XT (1)
hip: 5.4.22803-474e8620
xformers: unavailable
accelerate: 0.18.0
transformers: 4.26.1
Stable Diffusion: [cf1d67a] 2023-03-25
Taming Transformers: [9e9981c] 2022-11-07
CodeFormer: [c5b4593] 2022-09-09
BLIP: [48211a1] 2022-06-07
k_diffusion: [b43db16] 2023-02-06

https://github.com/vladmandic/automatic/commit/45dcc69484aea1cd78976dd8134dab6c7eb55546

**Comment by vladmandic** [comment]
this is not about cross-optimizations, this is your system decided to use cpu instead of gpu.
and you can see in profiling output which captures cpu usage - `txt2img` call should not be ~20sec if you have any kind of gpu.

you've mentioned you're using amd with cuda hip - what does webui say in the console on startup? is it detected?

**Comment by CapsAdmin** [comment]
It doesn't say anything indicating that it's using CPU as far as I can see.

```
21:12:10-863138 INFO     Python 3.10.6 on Linux                                                                                                                                                              
21:12:10-869511 INFO     No changes detected: quick launch active                                                                                                                                            
21:12:10-870242 INFO     Setting environment tuning                                                                                                                                                          
21:12:11-726111 INFO     Torch 2.1.0.dev20230415+rocm5.4.2                                                                                                                                                   
21:12:11-990134 INFO     Torch backend: AMD ROCm HIP 5.4.22803-474e8620                                                                                                                                      
21:12:11-991501 INFO     Torch detected GPU: AMD Radeon RX 6900 XT VRAM 16368 Arch (10, 3) Cores 40                                                                                                          
21:12:11-992209 INFO     Server arguments: ['--profile']                                                                                                                                                     
No module 'xformers'. Proceeding without it.
```

**Comment by CapsAdmin** [comment]
I should also mention that I can see the GPU is utilized when rendering, though not by much.

**Comment by CapsAdmin** [comment]
I've setup a venv that I share between this, a1111 original and comfyui. It seems to only be slow in this repo somehow.

**Comment by xedis** [comment]
I could be totally wrong here but..

> I've setup a venv that I share between this, a1111 original and comfyui. It seems to only be slow in this repo somehow.

There are significant version differences in deps between automatic1111 and vlad's fork, I cant speak to comfyui but I imagine the same is true. 

You should try setting up separate venv's for each, as this is why we use venv to begin with.

Personally I just create separate venv's in each cloned repo.

**Comment by vladmandic** [comment]
yup, it may not cause visible problems, but difference in dependencies is quite large.
and you can run into "i'm not really sure which version of the module" i'm using when you start a1111 or this repo since both do some requirements checks and overwrites on startup, but not all (due to performance).

**Comment by CapsAdmin** [comment]
I did this to exclude packages not being the culprit where this repo had precedent. (commented out the auto upgrade logic in a1111) However I tried my a1111 venv that I had backed up, and it worked fine in this fork (though it did update some of the packages on startup)

**So now my problem is resolved**, but sadly I'm not sure how to reproduce it. It seems like one dependency somewhere that does not pin the version could be the cause. I'm very confident the GPU was being used as pytorch reports that it can see my GPU, I can see the GPU being utilized and explicitly running with the  CPU mode flag doesn't even work on my pc as it erros about device types and other things.

Maybe issues flagged as questions could be moved to discussions?
